### ISSUE TITLE
Add Auto-augment input transformation

### ISSUE THREAD

**Issue by datumbox** [pytorch/vision#3050]
## 🚀 Feature
Provide an Implementation of the [AutoAugment](https://arxiv.org/pdf/1805.09501.pdf) policies for ImageNet, CIFA10 and SVHN on TorchVision.

## Motivation

AutoAugment is a common Data Augmentation technique that when applied, typically increases the accuracy of the models on specific datasets. On the original paper, the authors used RL to search for better data augmentation policies that increase the accuracy on 3 datasets: ImageNet, CIFA10 and SVHN. Though the found data augmentation policies were directly linked to their trained dataset, the paper shows that ImageNet policies provide significant improvements when applied to other datasets such as Oxford Flowers, Caltech-101, Oxford-IIT Pets, FGVC Aircraft and Stanford Cars. Similarly the CIFAR10 policies can be used for boosting the performance of CIFAR100.

Hence providing an out-of-the-box implementation for the aforementioned policies can be beneficial for the users of TorchVision.

## Pitch

Provide the implementation of 3 the following policies:

- [x] ImageNet AutoAugment Policy
- [x] CIFAR10 AutoAugment Policy
- [x] SVHN AutoAugment Policy

Currently there are already [community implementations](https://github.com/DeepVoltaire/AutoAugment/) for the above, nevertheless they all use PIL as backend. Therefor to add this in TorchVision we need to add support for the following missing Tensor Functional transforms:

- [x] posterize
- [x] solarize
- [x] equalize
- [x] sharpness
- [x] autocontrast
- [x] invert

Note that the original paper also experimented with the following augmentation techniques:

1. [Cutout](https://arxiv.org/pdf/1708.04552.pdf): Very similar to [RandomErasing](https://arxiv.org/pdf/1708.04896.pdf) but with does not apply thresholds on rations/scales, can produce multiple rectangles, uses the value of the average pixel per channel of the dataset. TorchVision already provides an [implementation](https://github.com/pytorch/vision/blob/master/torchvision/transforms/transforms.py#L1389-L1413) for RandomErasing which closely approximates Cutout, so it's probably not worth reimplementing it. Moreover though cutout was investigated by AutoAugment's RL as a policy, it was not selected for any of the datasets, though it was applied always at the end on the CIFAR10 experiments.
2. [Sample Pairing](https://arxiv.org/pdf/1801.02929.pdf): Similar to the above, though Sample Pairing was investigated during the RL phase, it was not selected by any of the dataset policies and thus we don't need to implement it here.
 

Lastly it's important to note that it's worth supporting the aforementioned transforms even if we decide not to implement AutoAugment. This is because it will enable the easier addition of newer Data Augmentation techniques such as [FixMatch](https://arxiv.org/pdf/2001.07685.pdf) ([official implementation](https://github.com/google-research/fixmatch), @vfdev-5's [implementation](https://github.com/vfdev-5/FixMatch-pytorch)), [CTAugment](https://arxiv.org/pdf/1911.09785.pdf) etc.

cc @vfdev-5 @fmassa
